### DISCUSSION TITLE
Convert SixLabors.ImageSharp.Image to System.Drawing.Image?

### DISCUSSION THREAD

**Discussion by SoftCircuits** [SixLabors/ImageSharp#1355] [Q&A] [answered]
I have a WinForms application with a PictureBox control. Is there any way to convert a `SixLabors.ImageSharp.Image` to a `System.Drawing.Image` short of writing it to a file and reloading it?

FWIW, my only interest here is in resizing an image.

**Comment by JimBobSquarePants** [comment, marked as answer]
Check out this class from the test utilities. It contains methods to copy pixel data to/from ImageSharp images to System.Drawing ones. 

https://github.com/SixLabors/ImageSharp/blob/b06cb32b7114961fd5473f7645d38f8fee04ec64/tests/ImageSharp.Tests/TestUtilities/ReferenceCodecs/SystemDrawingBridge.cs

**Reply by antonfirsov** [reply]
@SoftCircuits you need the method `To32bppArgbSystemDrawingBitmap()` there. 

Note that it's code can be simplified a lot by removing the generic argument from the method and fixing the pixel type to `Bgra32`. The pixel data of an `Image<Bgra32>`  can be directly copied to a `Bitmap` of `PixelFormat.Format32bppArgb`. (The difference between the component order is not a typo!) 

In that case both `workBuffer` and the following call is becomes unnecessary:
https://github.com/SixLabors/ImageSharp/blob/b06cb32b7114961fd5473f7645d38f8fee04ec64/tests/ImageSharp.Tests/TestUtilities/ReferenceCodecs/SystemDrawingBridge.cs#L159

Instead of the conversion you can just copy `Span<TPixel> row` to the buffer starting at `destPtr`.
